propagation-stackdriver/src/main/java/zipkin2/propagation/stackdriver/XCloudTraceContextExtractor.java

```diff
@@ -52,24 +53,20 @@ public TraceContextOrSamplingFlags extract(C carrier) {
 
       // Try to parse the trace IDs into the context
       TraceContext.Builder context = TraceContext.newBuilder();
+      long[] traceId = convertHexTraceIdToLong(tokens[0]);
+      if (traceId != null) {
+        boolean traceTrue = true;
+
+        if (tokens.length >= 2) {
+          int semicolonPos = tokens[1].indexOf(";");
+          traceTrue = semicolonPos == -1
+                  || tokens[1].length() == semicolonPos + 4
+                  && tokens[1].charAt(semicolonPos + 3) == '1';
+        }
 
-      if (tokens.length >= 2) {
-        long[] traceId = convertHexTraceIdToLong(tokens[0]);
-        int semicolonPos = tokens[1].indexOf(";");
-        String spanId = semicolonPos == -1 ? tokens[1] : tokens[1].substring(0, semicolonPos);
-        boolean traceTrue =
-            semicolonPos == -1
-                || tokens[1].length() == semicolonPos + 4
-                    && tokens[1].charAt(semicolonPos + 3) == '1';
-
-        if (traceId != null && traceTrue) {
-          result =
-              TraceContextOrSamplingFlags.create(
-                  context
-                      .traceIdHigh(traceId[0])
-                      .traceId(traceId[1])
-                      .spanId(Long.parseLong(spanId))
-                      .build());
+        if (traceTrue) {
+          result = TraceContextOrSamplingFlags.create(
+                  context.traceIdHigh(traceId[0]).traceId(traceId[1]).spanId(1L).build());
```

Comment:
I'm not sure about the opportunistic parsing because the span ID in Stackdriver is by definition 128-bit. So, overflowing 64 bits is not exactly an edge case.
See: https://cloud.google.com/trace/docs/support

Reply:
@adriancole Sorry, I take back my comment. The span ID is actually 64-bit, but it's unsigned. Hence, it fails about 50% of the time. We just need to deal with it correctly.

Reply:
```
[INFO] Checking unresolved references to org.codehaus.mojo.signature:java16:1.0
[ERROR] /usr/local/google/home/joaomartins/workspaces/zipkin-gcp/propagation-stackdriver/src/main/java/zipkin2/propagation/stackdriver/XCloudTraceContextExtractor.java:75: Undefined reference: long Long.parseUnsignedLong(String)
```

Any reason we're sticking to Java 1.6 here? Or can I change it?